rfd/0073-discover.md

```diff
@@ -0,0 +1,203 @@
+links to the respective sections of the documentation guides for preparing
+the node (e.g. joining to AD domain for SQL Server) and show the commands for
+creating a database user.
+
+For AWS databases that use IAM authentication (PostgreSQL, MySQL), the wizard
+will additionally display an appropriate `teleport db configure bootstrap`
+command for the user to run which will configure IAM permissions.
+
+### Step 4. Start agent
+
+For SSH, display the command for the user to run on their node, showing either
+"Automatic" or "AWS" commands from the existing "Add server" dialog depending
+on whether self-hosted or AWS option was picked in step 1.
+
+For a database, display the appropriate `teleport db start` command similar to
+the existing "Add database" dialog.
+
+### Step 5. Configure role
+
+Teleport roles by default only include internal user traits as allowed SSH
+logins (`{{internal.logins}}`) and database users (`{{internal.db_users}}`)
+and database names (`{{internal.db_names}}`). This results into users getting
+access denied errors unless they update their roles explicitly.
+
+On this step the wizard will ask the user for their intended SSH logins and/or
+database users/names and update the internal user traits appropriately so the
+users with the built-in `access` role will be allowed access to their resources.
+
+### Step 6. Connect
+
+This step will ask the user to test the connectivity by logging into their
+cluster with `tsh login` and running appropriate connect command, `tsh ssh`
+or `tsh db connect`.
+
+### Day 2
+
+Day 2 users already have gotten an initial success with connecting their
+resources to Teleport by going through the guided wizard described above.
+They have SSH and/or database agent(-s) installed and running.
+
+As they're thinking about bringing a larger part of their infrastructure into
+their Teleport cluster, this is where it makes sense for them to use Teleport's
+auto-discovery mechanisms.
+
+Currently, the auto-discovery can only be configured by updating the static
+Teleport agent configuration file `teleport.yaml`:
+
+```yaml
+ssh_service:
+  enabled: "yes"
+  aws:
+  - types: ["ec2"]
+    regions: ["us-west-1"]
+    tags:
+      "*": "*"
+db_service:
+  enabled: "yes"
+  aws:
+  - types: ["rds"]
+    regions: ["us-west-1"]
+    tags:
+      "*": "*"
+```
+
+Instead, Teleport will implement ability to configure auto-discovery (enable,
+disable, specify resources types to discover, tags, etc.) dynamically via the
+API which web UI will utilize.
+
+Similar to application and database dynamic registration, auto-discovery
+configuration will be turned into a resource (e.g. `kind: Discovery`) which
+will be tied to a particular SSH or a database agent.
+
+Web UI will provide a wizard-like dialog that will allow users to enable AWS
+auto-discovery by going through the following flow:
+
+1. Select the type of resource to discover e.g. EC2 instances, RDS databases,
```

Comment:
I think this flow looks good, but needs a bit more detail on specifics of each stage.